docs/mcd.md

```diff
@@ -0,0 +1,326 @@
+
+### `<selection-group>`
+
+A required child of field when value-type is `option`. This is the container element for the options that will appear in the drop-down. It has no XML attributes. 
+
+Multiple selection-group elements can be used; see `<conditions>`.
+
+### `<option>`
+
+A child of selection-group, this represents one entry in the drop-down. It has the following XML attributes. 
+
+| Name  | Meaning | Optional? | Value Notes | Other Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| value | The value sent to ConnectionBuilder() | No | Any string value | |
+| label | The text the user sees | No | Any string value | |
+
+### `<conditions>`
+
+An optional child of field or selection group, this is used for conditional display. It specifies whether the field or selection group should be visible in the dialog based on the values the user has specified for other fields. 
+
+Multiple condition elements can be used, and will be OR'd. That is, the field or selection group will be visible when any one of the `<condition>` elements is matched. The connector will not load if there are circular references. 
+
+Hence, this is a container element for the condition elements. It has no XML attributes. 
+
+### `<condition>`
+
+| Name  | Meaning | Optional? | Value Notes | Other Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| field | Name of the field whose value will be checked for equality | No | Any string value | |
+| value | The value to match. If the field's value is equal to this, the condition is true. | No | Any string value | |
+
+
+## Example
+
+The Connection Fields file below demonstrates both conditional display of fields and a non-editable field. 
+
+The images below show the Connection Dialog produced using this file. The left side shows the dialog as it is initially displayed, and the right side shows what it changes to when the user selects Authentication Kerberos. 
+
+<p float="left">
+  <img src="{{ site.baseurl }}/assets/mcd-connection-dialog-1.png" alt="Connection Dialog as initially displayed" align=top hspace=40/>
+  <img src="{{ site.baseurl }}/assets/mcd-connection-dialog-2.png" alt="Connection Dialog after user selects Authentication Kerberos" align=top />
+</p>
+
+AutoReconnect is an example of a non-editable field. It will not be visible in the connection dialog, but its default value of `0` will be available in ConnectionBuilder() to add to the Connection String.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+ 
+<connection-fields>
+
+  <field name="server" label="Server" category="endpoint" value-type="string">
+    <validation-rule reg-exp="^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"/>
+  </field>
+ 
+  <field name="port" label="Port" category="endpoint" value-type="string" default-value="1000"/>
+ 
+  <field name="authentication-type" label="Authentication" category="authentication" value-type="selection" >
```

Comment:
While the most complete, I worry about the hive dialog example being overly complicated (and also not necessarily what we'd recommend today).

For example, I think out guidance should be to always use 'authentication' with certain values for platform support:
https://gitlab.tableausoftware.com/connectivity/connector-plugin-sdk-internal/blob/master/docs/attributes.md#authentication  

Probably only document auth-none and auth-user-pass for the moment, but let's compare with this page as well: https://tableau.github.io/connector-plugin-sdk/docs/auth-modes

Reply:
I'll replace the example by a simpler one, without the kerberos and Microsoft Azure Hindsight entries, without the transport type options, and without the SSL options; so basically we'll just show the username and password fields depending on which authentication type the user selects. Sound right?